libs/sdk-ui-dashboard/src/filterBar/DashboardDateFilter.tsx

```diff
@@ -12,15 +15,35 @@ export interface IDashboardDateFilterProps {
     /**
      * Definition of filter to render.
      */
-    filter: IDashboardDateFilter;
+    filter: IDashboardDateFilter | undefined;
 
     /**
      * When the user interacts with the filter and changes its value, it MUST use this callback to propagate the
      * new filter value.
      *
      * @param filter - new date filter value.
      */
-    onFilterChanged: (filter: IDashboardDateFilter) => void;
+    onFilterChanged: (filter: IDashboardDateFilter | undefined) => void;
+
+    /**
+     * Granularities available in the Floating range form.
+     */
+    availableGranularities: DateFilterGranularity[];
+
+    /**
+     * A {@link @gooddata/sdk-ui-filters#IDateFilterOptionsByType} configuration of the Date Filter.
+     */
+    dateFilterOptions: IDateFilterOptionsByType;
+
+    /**
+     * Optionally specify custom filter name. Defaults to "Date range" (or its localized equivalent).
+     */
+    customFilterName?: string;
```

Comment:
i propose to wrap these into an object that would encapsulate the config that should be applied on the date filter.

also, contemplating that perhaps that config needs to land already in props of the filter bar as an optional part of the contract. "filter bar should render these date filter and attribute filters. the date filter should have these settings"

Reply:
Moved to anothe type. I am not sure that it is the cleanest solution, but let's revisit that later?